dhall-lsp-server/src/Dhall/LSP/Backend/Linting.hs

```diff
@@ -6,46 +6,36 @@ module Dhall.LSP.Backend.Linting
 where
 
 import Dhall.Parser (Src)
-import Dhall.Core (Expr(..), Binding(..), Var(..), subExpressions, freeIn, Import)
+import Dhall.Core (Expr(..), Var(..), subExpressions, freeIn, Import)
 import qualified Dhall.Lint as Dhall
 
 import Dhall.LSP.Backend.Diagnostics
 
+import qualified Data.Maybe
 import Data.Monoid ((<>))
 import Data.Text (Text)
-import Data.List.NonEmpty (NonEmpty(..), tails, toList)
 import Control.Lens (universeOf)
 
 data Suggestion = Suggestion {
     range :: Range,
     suggestion :: Text
     }
 
--- Diagnose nested let blocks.
-diagLetInLet :: Expr Src a -> [Suggestion]
-diagLetInLet (Note _ (Let _ (Note src (Let _ _)))) =
-  [Suggestion (rangeFromDhall src) "Superfluous 'in' before nested let binding"]
-diagLetInLet _ = []
```

Comment:
As documented in https://github.com/dhall-lang/dhall-haskell/pull/1242/files#diff-281e7ffb783d37c08218e547d14115b2R155, we can use `Note`s to diagnose let-in-let. I think we just have to take care not to make multiple suggestions for the same let.